src/page_scrapers/parse_question_answers_page.ts

```diff
@@ -0,0 +1,76 @@
+import { Log } from "apify";
+import { combineUrl } from "../helpers/combine_url.js";
+import { parseJsonContent } from "../helpers/parse_json_content.js";
+import { unixToDateIso } from "../helpers/unixToDateIso.js";
+import { AnswerInfo } from "../types/parser_results.js";
+
+// endpoint for question answers returns some other entities, so it's needed to filter by typename
+const ANSWER_TYPENAME = "QuestionAnswerItem";
+
+// API responses typed as any deliberately because API is complex and subject to change
+export const parseQuestionAnswersPage = (
+    result: any,
+    log: Log
+): {
+    answers: AnswerInfo[];
+    pageInfo: {
+        hasNextPage: boolean;
+        endCursor: string;
+    };
+} => {
```

Comment:
I would declare the type of the function return value separately, this makes it a bit hard to read for me. Usually, we do it for all complex types (more complex than e.g. string[]).